### PR TITLE
[DF] Minor misc fixes to Snapshot and its docs

### PR DESCRIPTION
**Pull request by eguiraud**

### CODE REVIEW

File: tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx [hunk 1]
```diff
@@ -1340,15 +1341,14 @@ public:
...
-         // because TTree::Write writes in gDirectory, not in fDirectory
-         fOutputTree->AutoSave("flushbaskets");
-         // must destroy the TTree first, otherwise TFile will delete it too leading to a double delete
-         fOutputTree.reset();
-         fOutputFile->Close();
-      } else {
-         Warning("Snapshot", "A lazy Snapshot action was booked but never triggered.");
-      }
+      R__ASSERT(fOutputTree != nullptr);
+      R__ASSERT(fOutputFile != nullptr);
```
Comment (bellenot) [inline tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx:1345]: Is `R__ASSERT` also valid for non-debug code? I.e. is it always evaluated?

**Review by bellenot — APPROVED** [review]
LGTM, and I trust you anyway

File: tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx (hunk 1 above)
Comment (eguiraud) [reply tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx:1345]: I hope not! :) These are debug asserts, if `Finalize` is called we really expect `fOutputTree` and `fOutputFile` to not be null (they are set in `Initialize`, a few lines above).

**Comment by eguiraud** [comment]
Rebased to fix conflicts, will merge if green

File: tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx (hunk 1 above)
Comment (bellenot) [reply tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx:1345]: OK, thanks for the clarification :-)

**Pull request merged**